main.tf

```diff
@@ -0,0 +1,67 @@
+module "s3_bucket" {
+  source = "git::https://github.com/plus3it/terraform-aws-tardigrade-s3-bucket.git?ref=5.0.0"
+
+  bucket        = var.bucket_name
+  force_destroy = true
+
+  policy = {
+    json = jsonencode({
+      "Version" : "2012-10-17",
+      "Statement" : [
+        {
+          "Effect" : "Allow",
+          "Principal" : {
+            "Service" : "acm-pca.amazonaws.com"
+          },
+          "Action" : [
+            "s3:PutObject",
+            "s3:PutObjectAcl",
+            "s3:GetBucketAcl",
+            "s3:GetBucketLocation"
+          ],
+          "Resource" : [
+            "arn:aws:s3:::${var.bucket_name}/*",
+            "arn:aws:s3:::${var.bucket_name}"
+          ],
+          "Condition" : {
+            "StringEquals" : {
+              "aws:SourceAccount" : "${data.aws_caller_identity.current.account_id}"
+            }
+          }
+        }
+      ]
+    })
+  }
+}
+
+resource "aws_acmpca_certificate_authority" "this" {
+
+  type = var.aws_acmpca_certificate_authority.type
+
+  certificate_authority_configuration {
+    key_algorithm     = var.private_certificate_authority.key_algorithm
+    signing_algorithm = var.private_certificate_authority.signing_algorithm
+
+    subject {
+      common_name = var.aws_acmpca_certificate_authority.common_name
+    }
+  }
+
+  dynamic "revocation_configuration" {
+    for_each = var.private_certificate_authority.revocation_configuration
+
+    content {
+      crl_configuration {
+        custom_cname       = var.private_certificate_authority.custom_cname
```

Comment:
There are levels of nested configuration here, that our own object should reflect. i.e.

```
var.private_certificate_authority.revocation_configuration.crl_configuration.custom_name
```

etc...